nix/flake.nix

```diff
@@ -0,0 +1,148 @@
+{
+  description = "A GPU-accelerated cross-platform terminal emulator and multiplexer written by @wez and implemented in Rust";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    freetype2 = {
+      url = "github:wez/freetype2/de8b92dd7ec634e9e2b25ef534c54a3537555c11";
+      flake = false;
+    };
+
+    harfbuzz = {
+      url = "github:harfbuzz/harfbuzz/60841e26187576bff477c1a09ee2ffe544844abc";
+      flake = false;
+    };
+
+    libpng = {
+      url = "github:glennrp/libpng/8439534daa1d3a5705ba92e653eda9251246dd61";
+      flake = false;
+    };
+
+    zlib = {
+      url = "github:madler/zlib/cacf7f1d4e3d44d871b605da3b647f07d718623f";
+      flake = false;
+    };
+  };
+
+  outputs = {
+    self,
+    flake-utils,
+    rust-overlay,
+    nixpkgs,
+    freetype2,
+    harfbuzz,
+    libpng,
+    zlib,
+    ...
+  }:
```

Comment:
This is mostly a style preference, but mentioning in case you don't know the pattern.  
You can use `{ self, ... }@inputs` so you don't need to spell out everything here, and it makes what's a flake input obvious below.